### PR TITLE
feat(authentication-service): added SAML example

### PR DESCRIPTION
**Pull request by karuneshsf**
## Description

added example for SAML

Fixes # (issue)
GH-1663
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Intermediate change (work in progress)

## How Has This Been Tested?
- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: sandbox/oauth-example/src/providers/saml-verify.provider.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      cb: samlStrategy.VerifiedCallback,
+      req?: Request,
+    ) => {
+      const user = await this.signupProvider(profile);
+      if (user == null) {
+        throw new HttpErrors.Unauthorized(AuthErrorKeys.InvalidCredentials);
+      }
+      // Just to check if userCreds has entry for this user for this provider
+      const creds = await this.userCredsRepository.findOne({
+        where: {
```
Comment (yeshamavani) [inline sandbox/oauth-example/src/providers/saml-verify.provider.ts:36]: ?? 
a verify provider should first check if user present or not and then call signup if not present

File: services/authentication-service/src/models/index.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ export * from './auth-secure-client.model';
 export * from './forget-password-dto.model';
 export * from './forget-password-response-dto.model';
 export * from './local-user-profile';
+export * from './login-activity.model';
```
Comment (yeshamavani) [inline services/authentication-service/src/models/index.ts:36]: revert this change
causes unnecessary release

Comment (karuneshsf) [reply services/authentication-service/src/models/index.ts:36]: This one is not change, the ordering of the file is only changed

File: sandbox/oauth-example/src/providers/saml-verify.provider.ts (hunk 1 above)
Comment (karuneshsf) [reply sandbox/oauth-example/src/providers/saml-verify.provider.ts:36]: This is being done in same manner in all other providers as well, so followed same pattern

File: sandbox/oauth-example/src/providers/google-oauth2-verify.provider.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      if (user == null) {
+        throw new HttpErrors.Unauthorized(AuthErrorKeys.InvalidCredentials);
+      }
+      // Just to check if userCreds has entry for this user for this provider
+      const creds = await this.userCredsRepository.findOne({
+        where: {
+          userId: user.id,
+          authProvider: 'google',
+        },
+      });
```
[the remark this answers is not in the document]
Comment (karuneshsf) [reply sandbox/oauth-example/src/providers/google-oauth2-verify.provider.ts]: Removed

File: sandbox/oauth-example/package.json [hunk 4]
```diff
@@ -60,6 +60,7 @@
     "@loopback/rest-explorer": "^6.1.3",
     "@loopback/service-proxy": "^6.1.3",
     "@sourceloop/authentication-service": "^14.1.1",
+    "loopback4-authentication": "^9.0.2",
```
[the remark this answers is not in the document]
Comment (karuneshsf) [reply sandbox/oauth-example/package.json]: updated

File: services/authentication-service/src/models/index.ts (hunk 2 above)
Comment (yeshamavani) [reply services/authentication-service/src/models/index.ts:36]: yes but still revert it 
will release the service with a minor version bump

**Review by yeshamavani — CHANGES_REQUESTED** [review]

**Comment by yeshamavani** [comment]
Please rebase

File: sandbox/oauth-example/src/models/auth-user.model.ts
Comment (yeshamavani) [inline sandbox/oauth-example/src/models/auth-user.model.ts:1]: why is this model needed 
we have auth user in authentication-service

Comment (karuneshsf) [reply sandbox/oauth-example/src/models/auth-user.model.ts:1]: removed it

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**